mixer/adapter/solarwinds/solarwinds.go

```diff
@@ -0,0 +1,115 @@
+// Copyright 2017 Istio Authors.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+//go:generate $GOPATH/src/istio.io/istio/bin/mixer_codegen.sh -f mixer/adapter/solarwinds/config/config.proto
+
+// Package solarwinds publishes metric and log values collected by Mixer
+// to appoptics and papertrail respectively.
+package solarwinds
+
+import (
+	"context"
+
+	"istio.io/istio/mixer/adapter/solarwinds/config"
+	"istio.io/istio/mixer/pkg/adapter"
+	"istio.io/istio/mixer/template/logentry"
+	"istio.io/istio/mixer/template/metric"
+)
+
+type (
+	builder struct {
+		cfg *config.Params
+	}
+
+	handler struct {
+		logger         adapter.Logger
+		metricsHandler metricsHandlerInterface
+		logHandler     logHandlerInterface
+	}
+)
+
+var (
+	_ metric.HandlerBuilder = &builder{}
+	_ metric.Handler        = &handler{}
+)
+
+// GetInfo returns the Info associated with this adapter.
+func GetInfo() adapter.Info {
+	return adapter.Info{
+		Name:        "solarwinds",
+		Impl:        "istio.io/istio/mixer/adapter/solarwinds",
+		Description: "Publishes metrics to appoptics and logs to papertrail",
+		SupportedTemplates: []string{
+			metric.TemplateName,
+			logentry.TemplateName,
+		},
+		NewBuilder:    func() adapter.HandlerBuilder { return &builder{} },
+		DefaultConfig: &config.Params{},
+	}
+}
+
+//func (b *builder) SetMetricTypes(map[string]*metric.Type) {}
+func (b *builder) SetAdapterConfig(cfg adapter.Config) {
+	// this is a common adapter config for both log and metric
+	b.cfg = cfg.(*config.Params)
+}
+
+func (b *builder) SetMetricTypes(map[string]*metric.Type) {}
+
+func (b *builder) SetLogEntryTypes(entries map[string]*logentry.Type) {}
+
+func (b *builder) Validate() *adapter.ConfigErrors { return nil }
```

Comment:
I think there are things to validate here. For example, batch size should be >=0 and <= 1000. Anything that can help the user write correct config is valuable.